packages/blade/src/components/Radio/_decisions/decisions.md

```diff
@@ -0,0 +1,113 @@
+# Radio Decisions <!-- omit in toc -->
+
+Radio is used to present each item in a list of options where users must make a single selection.
+
+This document outlines the API of `Radio` & `RadioGroup` component.
+
+<img src="./radio_thumbnail.png" alt="Radio component thumbnail" width="380" />
+
+- [Design](#design)
+- [`Radio` API](#radio-api)
+- [`RadioGroup` API](#radiogroup-api)
+  - [Examples:](#examples)
+    - [Basic](#basic)
+    - [Controlled, Uncontrolled](#controlled-uncontrolled)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+
+## Design
+
+- [Figma Link - Radio](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=13133%3A160709)
+
+## `Radio`
+
+| Prop                | Type                                    | Default     | Description                                                                                                                                                    | Required |
+| ------------------- | --------------------------------------- | ----------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- |
+| children            | `string`                                | `undefined` | The text to be rendered as RadioLabel                                                                                                                          | ✅       |
+| isChecked           | `boolean`                               | `false`     | state of the radio, if explicitly set the radio will become [controlled component](https://reactjs.org/docs/forms.html#controlled-components)                  |          |
+| defaultChecked      | `boolean`                               | `false`     | default state of the radio, if isChecked is not provided the radio will become [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html) |          |
+| onChange            | `({ event, isChecked, value }) => void` | `undefined` | The function to be called when the radio state changes.                                                                                                        |          |
```

Comment:
Do we need `event`? Also do we need both `isChecked` and `value` here 🤔

Reply:
+1, noticed this inconsistency in the Checkbox API where we are passing the event in Checkbox but not in the Checkbox group. Just thinking what possible use-cases will the consumer have with the event object in either cases (not sure if we have discussed this earlier). 

We will need both `isChecked` and `value`. `value` here refers to the value of the input field in the radio button, check [ref](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio#value)

Reply:
> noticed this inconsistency in the Checkbox API where we are passing the event in Checkbox but not in the Checkbox group

@nashcheez The reason is in a single Checkbox it has it's own `<input />` which can have `event` but CheckboxGroup and RadioGroups are abstractions which we built to provide state management and other features, thus it can't have an `event` object. 

> We will need both isChecked and value

Yes initially on the first draft I only kept `isChecked` but kamlesh suggested that for analytics and other services they could want the `value` too. [ref](https://github.com/razorpay/blade/pull/539#discussion_r925639702)

Reply:
@anuraghazra Noticed that for a single checkbox, we are not sending out `name` in the `onChange` handler (maybe, can be extracted from `event`, as `event.target.name`), but for `RadioGroup`, we are having `name` in the `onChange`. I know this PR is about `Radio`, but given `Radio` cannot exist w/o `RadioGroup` and we are sending out the name in `RadioGroup`, thought we should be doing the same for a single checkbox from analytics perspective. It will even make the analytics handler similar as it will always get the `{ name, value }` in the change handler.

Given how `Input` component's [decisions](https://github.com/razorpay/blade/blob/master/packages/blade/src/components/Input/_decisions/decisions.md) are taken, should we send out the `name` in single checkbox too for uniformity purpose in the `onChange` prop?

Reply:
Yeah sure we'll consider this and make the API consistent. For the time being we're freezing this API.